### PR TITLE
Final project submission

### PR DESCRIPTION
**Pull request by LeslieAine**
In this project, I create a budget app that allows a user to keep track of their expenses.

**Accomplishments:**

- Splash screen

- [ ] A simple page with the name of your app (yes, you need to choose one), and links to the sign up and log in pages.

- Sign up and log in pages

- [ ] The user should be able to register in the app with full name, email and password (all mandatory).
- [ ] The user can log into the app using email and password.
- [ ] If the user is not logged in, they can't access pages that require the user to be logged in (all the pages described below).

- Home page (categories page)

- [ ] When the user logs in, they are presented with the categories page.
- [ ] For each category, the user can see their name, icon and the total amount of all the transactions that belongs to that category.
- [ ] When the user clicks (or taps) on a category item, the application navigates to the transactions page for that category.
- [ ] There is a button "add a new category" at the bottom that brings the user to the page to create a new category.

- Transactions page

- [ ] For a given category, the list of transactions is presented, ordered by the most recent.
- [ ] At the top of the page the user could see the total amount for the category (sum of all of the amounts of the transactions in that category).
- [ ] There is a button "add a new transaction" at the bottom that brings the user to the page to create a new transaction.
- [ ] When the user clicks on the "Back" button (<), the user navigates to the home page.

- "Add a new category" page

- [ ] The user fills out a form to create a new category, indicating their name and icon (both mandatory).
- [ ] The user clicks (or taps) the "Save" button to create the new category, and is taken to the home page on success.
- [ ] When the user clicks on the "Back" button (<), the user navigates to the home page.
- [ ] "Add a new transaction" page

- The user fills out a form to create a new transaction with:

- [ ] name (mandatory)
- [ ] amount (mandatory)
- [ ] categories (mandatory at least one)
- [ ] The user click (or taps) the "Save" button to create the new transaction, and is taken to the transactions page for that category.
- [ ] When the user clicks on the "Back" button (<), the user navigates to the transactions page for that category.

- Testing requirements

- [ ] Create unit and integration tests for all the most important components of your RoR application.

### CODE REVIEW

File: config/database.yml [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+  <<: *default
+  database: budget_app_development
+  username: postgres
+  password: leslie
+
+  # The specified database role being used to connect to postgres.
+  # To create additional roles in postgres see `$ createuser --help`.
+  # When left blank, postgres will use the default role. This is
+  # the same name as the operating system user running Rails.
+  #username: budget_app
```
Comment (MahmoudBakr23) [inline config/database.yml:34]: - [ ] Please make sure to remove your own credentials for a professional code base. 🙏

**Review by MahmoudBakr23 — CHANGES_REQUESTED** [review]
Hello @LeslieAine 😃, 

Great work and effort you put into this project so far.

## Highlights 🚀 

There are some excellent points that you have made perfectly. Keep up! 💪
- Well done creating the database according to the ERD
- Well done working with the controllers, models, and views
- Good work with the unit and integration tests
- Nice presentation
- Descriptive PR and README
- Correct Linters

However, there is always room for improvement.

> I'd like to remind you that your long work hours are appreciated. However, the code review process is for improving your code, not criticizing it. 😄

 ## Required Changes ♻️

- [ ] Please make sure to deploy your app and attach the live demo link in the README file

_Please check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code._

N/A

Cheers, and Happy coding!👏👨‍💻💻

Feel free to mention @MahmoudBakr23 to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: config/database.yml (hunk 1 above)
Comment (LeslieAine) [reply config/database.yml:34]: Thank you @MahmoudBakr23 for your time to look at my code. Appreciate it 🙏🏽

**Review by Reem-lab — APPROVED** [review]
# Hi @LeslieAine,  👋🏻👋🏻

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


## Highlights

- [ ] Great job on the funtionality for the setup with devise 💯
- [ ]  ERD is set up well with the migrations ⭐
- [ ]  Great job working on the functionality 🔥
- [ ]  Followed Gitflow 🔥
- [ ]  Goodworking on design ✨✨💯💯
- [ ]  No Linters error ✅
- [ ]  Kudos on the huge efforts for the tests, All passed ✅✅
- [ ]  Professional README file 🔥🔥
- [ ]  Deployed well 💯💯


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by LeslieAine** [comment]
Thank you @Reem-lab for the time to look at my code and the suggestions. 🙏🏽

**Pull request merged**